akka-javasdk/src/main/scala/akka/javasdk/impl/SdkRunner.scala

```diff
@@ -660,6 +664,15 @@ private final class Sdk(
       healthCheck = () => SdkRunner.FutureDone)
   }
 
+  private def isDisabled(disabledComponents: Set[Class[_]])(componentDescriptor: spi.ComponentDescriptor): Boolean = {
+    val className = componentDescriptor.implementationName
+    if (disabledComponents.map(_.getName).contains(className)) {
```

Comment:
move `disabledComponents.map(_.getName)` to the call site so that it's only done once, i.e.
```
def isDisabled(disabledComponents: Set[String]
```

Reply:
fixed